tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/FilteredData.java

```diff
@@ -0,0 +1,67 @@
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package org.unicode.icu.tool.cldrtoicu;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import javax.annotation.Nullable;
+
+import org.unicode.cldr.api.CldrData;
+import org.unicode.cldr.api.CldrPath;
+import org.unicode.cldr.api.CldrValue;
+
+/**
+ * A class which allows data from some underlying {@link CldrData} source to be filtered or
+ * removed (but not added).
+ */
+// TODO: Once DTD ordering is the only allowed order, this can be extended to allow adding paths.
+abstract class FilteredData implements CldrData {
```

Comment:
Base class representing filtered CLDR data (used by both alt-path and pseudo locales).